File: bukkit/elysium-core-bukkit/src/main/java/com/seventh_root/elysium/core/bukkit/listener/PluginEnableListener.java
package com.seventh_root.elysium.core.bukkit.listener;

import com.seventh_root.elysium.core.bukkit.ElysiumCoreBukkit;
import com.seventh_root.elysium.core.bukkit.plugin.ElysiumBukkitPlugin;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.server.PluginEnableEvent;

public class PluginEnableListener implements Listener {

    private final ElysiumCoreBukkit plugin;

    public PluginEnableListener(ElysiumCoreBukkit plugin) {
        this.plugin = plugin;
    }

    @EventHandler
    public void onPluginEnable(PluginEnableEvent event) {
        if (event.getPlugin() != plugin) {
            if (event.getPlugin() instanceof ElysiumBukkitPlugin) {
                ElysiumBukkitPlugin elysiumBukkitPlugin = (ElysiumBukkitPlugin) event.getPlugin();
                plugin.initializePlugin(elysiumBukkitPlugin);
            }
        }
    }

}

File: bukkit/elysium-core-bukkit/src/main/kotlin/com/seventh_root/elysium/core/bukkit/listener/PluginEnableListener.kt
package com.seventh_root.elysium.core.bukkit.listener

import com.seventh_root.elysium.core.bukkit.ElysiumCoreBukkit
import com.seventh_root.elysium.core.bukkit.plugin.ElysiumBukkitPlugin
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.server.PluginEnableEvent

class PluginEnableListener(private val plugin: ElysiumCoreBukkit): Listener {

    @EventHandler
    fun onPluginEnable(event: PluginEnableEvent) {
        if (event.plugin !== plugin) {
            if (event.plugin is ElysiumBukkitPlugin) {
                val elysiumBukkitPlugin = event.plugin as ElysiumBukkitPlugin
                plugin.initializePlugin(elysiumBukkitPlugin)
            }
        }
    }

}
